cunumeric/array.py

```diff
@@ -184,6 +186,140 @@ def _convert_all_to_numpy(obj: Any) -> Any:
+    def __init__(
+        self,
+        c_contiguous: bool = True,
+        f_contiguous: bool = False,
+        owndata: bool = True,
+        writeable: bool = True,
+        aligned: bool = True,
+        writebackifcopy: bool = False,
+    ) -> None:
+        self._c_contiguous = c_contiguous
+        self._f_contiguous = f_contiguous
+        self._owndata = owndata
+        self._writeable = True
+        self._aligned = True
+        self._writebackifcopy = False
+
+    def __getitem__(self, key: Any) -> bool:
+        key = self.check_flag(key)
+        return getattr(self, key)
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        key = self.check_flag(key)
+        setattr(self, key, value)
+
+    def __repr__(self) -> str:
+        output = ""
+        for each in self.__dict__:
+            output += f"{each.strip('_').upper()} : {self.__dict__[each]} \n"
+        return output
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def check_flag(self, key: str) -> str:
+        key = key.lower()
+        if len(key) == 1:
+            if key == "x":
+                key = "writebackifcopy"
+            else:
+                for attr in self.__dict__.keys():
+                    if (str)(attr).strip("_")[0] == key:
+                        key = (str)(attr)
+                        break
+
+        if not hasattr(self, key):
+            raise AttributeError(f"{key} is unknown flag for ndarray.flags")
+        return key
+
+    @property
+    def c_contiguous(self) -> bool:
+        return self._c_contiguous
+
+    @property
+    def f_contiguous(self) -> bool:
+        return self._f_contiguous
+
+    @property
+    def owndata(self) -> bool:
+        return self._owndata
+
+    @property
+    def writeable(self) -> bool:
+        return self._writeable
+
+    @property
+    def aligned(self) -> bool:
+        return self._aligned
+
+    @property
+    def writebackifcopy(self) -> bool:
+        return self._writebackifcopy
+
+    @property
+    def fnc(self) -> bool:
+        return (bool)(self.f_contiguous and ~self.c_contiguous)
+
+    @property
+    def forc(self) -> bool:
+        return (bool)(self.f_contiguous or self.c_contiguous)
+
+    @property
+    def behaved(self) -> bool:
+        return (bool)(self.aligned and self.writeable)
+
+    @property
+    def carray(self) -> bool:
+        return (bool)(self.behaved and self.c_contiguous)
+
+    @property
+    def farray(self) -> bool:
+        return (bool)(
+            self.behaved and self.f_contiguous and ~self.c_contiguous
+        )
+
+    @writeable.setter  # type: ignore
+    def writeable(self, value: bool) -> None:
+        if value is True and self.owndata is False:
+            raise ValueError(
+                "cannot set WRITEABLE flag to True for this array"
+            )
+        self._writeable = value
+
+    @aligned.setter  # type: ignore
+    def aligned(self, value: bool) -> None:
+        self._aligned = value
```

Comment:
For `value = True`, should we check if self is truly aligned ?

Reply:
I think it should be considered separately. For now, we keep this interface and later we can add some logic to check if the underlying data is aligned. Maybe, legate should give some info about this.

Reply:
then should we remove _aligned flag at the moment? Since we are not checking it correctly?

Reply:
We can remove setter then. Flag should exist. We can set `False`  if we're not sure and not allowing the value to be True at this moment. But, the flag should exist. As I said, I have no idea how to get information of alignment of data here.

Reply:
Let's make it False for now and remove the setter